aspnetcore/mvc/views/razor.md

```diff
@@ -512,92 +581,157 @@ If "rick@contoso.com" is passed in the model, the view generates the following H
+
+If the *EvenMorePages* folder in the preceding example has an imports file with `@namespace Another.Planet` (or the *Pages/MorePages/EvenMorePages/Page.cshtml* file contains `@namespace Another.Planet`), the result is shown in the following table.
+
+| Page                                        | Namespace               |
+| ------------------------------------------- | ----------------------- |
+| *Pages/Index.cshtml*                        | `Hello.World`           |
+| *Pages/MorePages/Page.cshtml*               | `Hello.World.MorePages` |
+| *Pages/MorePages/EvenMorePages/Page.cshtml* | `Another.Planet`        |
+
+### \@page
+
+::: moniker range=">= aspnetcore-3.0"
+
+The `@page` directive has different effects depending on the type of the file where it appears. The directive:
+
+* In in a *.cshtml* file indicates that the file is a Razor Page. For more information, see <xref:razor-pages/index>.
+* Specifies that a Razor component should handle requests directly. For more information, see <xref:blazor/routing>.
 
 ::: moniker-end
 
-### @attribute
+::: moniker range="< aspnetcore-3.0"
 
-The `@attribute` directive adds the given attribute to the class of the generated page or view. The following example adds the `[Authorize]` attribute:
+The `@page` directive turns a file into an MVC action, which means that the file handles requests directly, without going through a controller. For more information, see <xref:razor-pages/index>.
 
-```cshtml
-@attribute [Authorize]
-```
+::: moniker-end
 
-### @namespace
+### \@section
 
-The `@namespace` directive sets the namespace of the class of the generated page or view:
+*This scenario only applies to MVC views and Razor Pages (.cshtml).*
 
-```cshtml
-@namespace Your.Namespace.Here
-```
+The `@section` directive is used in conjunction with [MVC and Razor Pages layouts](xref:mvc/views/layout) to enable views or pages to render content in different parts of the HTML page. For more information, see <xref:mvc/views/layout>.
+
+### \@using
+
+The `@using` directive adds the C# `using` directive to the generated view:
+
+[!code-cshtml[](razor/sample/Views/Home/Contact9.cshtml)]
+
+::: moniker range=">= aspnetcore-3.0"
+
+In [Razor components](xref:blazor/components), `@using` also controls which components are in scope.
+
+::: moniker-end
+
+::: moniker range=">= aspnetcore-3.0"
+
+## Directive attributes
+
+*Directive attribute scenarios in this section only apply to Razor components (.razor).*
+
+### \@attributes
+
+*This scenario only applies to Razor components (.razor).*
 
-If a page or view imports API with an `@namespace` directive, the original file's namespace is set relative to that namespace. 
+A component can capture and render additional attributes in addition to the component's declared parameters using the `@attributes` Razor directive. For more information, see <xref:blazor/components#attribute-splatting-and-arbitrary-parameters>.
```

Comment:
`@attributes`  allows a component to render non-declared attributes. For more information, see <xref:blazor/components#attribute-splatting-and-arbitrary-parameters>.